dcicutils/deployment_utils.py

```diff
@@ -18,20 +18,45 @@ def main():
 
 """
 
+import argparse
 import datetime
 import glob
 import io
 import json
+import logging
 import os
 import pkg_resources
 import re
 import subprocess
 import sys
 import toml
-import argparse
 
-from dcicutils.env_utils import get_standard_mirror_env, data_set_for_env, get_bucket_env, INDEXER_ENVS
-from dcicutils.misc_utils import PRINT
+from .env_utils import get_standard_mirror_env, data_set_for_env, get_bucket_env, INDEXER_ENVS
+from .misc_utils import PRINT
+
+
+def boolean_setting(settings, key, default=None):
+    """
+    Given a setting from a .ini file and returns a boolean interpretation of it.
+    - Treats an actual python boolean as itself
+    - Using case-insensitive comparison, treats the string 'true' as True and both 'false' and '' as False.
+    - If an element is missing,
+    - Treats None the same as the option being missing, so returns the given default.
+    - Raises an error for any non-standard value. This is harsh, but it should not happen.
+    """
+    if key not in settings:
+        return default
+    setting = settings[key]
+    if isinstance(setting, str):
+        setting_lower = setting.lower()
+        if setting_lower in ("", "false"):
+            return False
+        elif setting_lower == "true":
+            return True
+        else:
+            return setting
+    else:  # booleans, None, and odd types (though we might want to consider making other types an error).
```

Comment:
Should we not coerce None into False?

Reply:
Since None is already a false value, I thought it might be better to leave it as None in case we have to debug the difference between an explicit value and no value. I could change that if you thought it wise ,but my sense was it was better to leave the thin distinction.

Reply:
Yeah I think that's a good point, so let's keep it.